AirCasting/Services/DownloadMeasurmentsService.swift

```diff
@@ -67,6 +70,14 @@ final class DownloadMeasurementsService: MeasurementUpdatingService {
                     } catch {
                         assertionFailure("Failed to save context \(error)")
                     }
+                    persistenceController.performBackgroundTask { [weak self] bgContext in
+                        do {
+                            try self?.removeOldService.removeOldestMeasurements(in: bgContext,
+                                                                                uuid: uuid)
```

Comment:
please use capture list to capture strongly removeOldService and not weakly self. The logic should always happened even if self would delicate. 
```
 persistenceController.performBackgroundTask { [removeOldService] bgContext in
    do { 
      try removeOldService.removeOldestMeasurements(in: bgContext,                                                                                uuid: uuid)
}
```

Reply:
Uh, I couldn't do that. Errors everywhere. 
But I found this solution, is that better than `[weak self]`?
 
```
    private func updateForSession(uuid: SessionUUID) {
        let syncDate = Date().addingTimeInterval(-100)
        lastFetchCancellableTask = fixedSessionService.getFixedMeasurement(uuid: uuid, lastSync: syncDate, completion: { [removeOldService, persistenceController] result in
```